### DISCUSSION TITLE
Denkovi V2 4-relay board not responding

### DISCUSSION THREAD

**Discussion by vinnio** [theyosh/TerrariumPI#638] [Q&A] [answered]
Hi,

I have 2 Denkovi V2 4-relay boards connected to my Raspberry 4B. Using your instructions, I have installed Java (11 and 8, tried both).
I have placed the CommandLineTool in the 3rdparty-folder.  

```
java -jar 3rdparty/DenkoviRelayCommandLineTool/DenkoviRelayCommandLineTool.jar ID=0 4v2 1 1`
1000
```
doesn't throw any errors, but it doesn't change the state of relay 1. Neither for ID=1 or other relays (1,2,3,4 of both boards).

`java -jar 3rdparty/DenkoviRelayCommandLineTool/DenkoviRelayCommandLineTool.jar list`
shows a list of 2 relay-boards, so the commandline-tool is working, but it can't alter the state of the relays.

When tested on Windows 10 PC, with the DRMv3 software, the relays can be used and switched without problems. 
I've mailed Denkovi as well, but do you know how this can be fixed? Did I miss anything? 

Some more information:

```
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
Linux raspberrypi 5.10.88-v7l+ #1504 SMP Wed Dec 22 14:21:11 GMT 2021 armv7l GNU/Linux
Model           : Raspberry Pi 4 Model B Rev 1.1
```

```
OS: com.sun.jna.Platform.OSType=1, com.sun.jna.Platform.ARCH=arm, java.lang.System.os.name=Linux;
        Denkovi Relay Command Line Tool v2.14 / 24 Nov 2021
        Libraries: DenkoviHID-1.6, JavaFTD2XX-1.4d, JNA-5.6.0;
        Contact: support@denkovi.com
        Web site: http://denkovi.com/denkovi-relay-command-line-tool
        Developer: Denkovi Assembly Electronics LTDRelay Command Line Tool
```

But even after normal testing using JAVA and using the command-line tool, they won't turn on/off. For both of the boards. Used ID =1 and =0, tried 0005771479 and the other serial number. To no avail.. 

```
> java -jar DenkoviRelayCommandLineTool.jar list
Serial number [Description] [ID]
FTDI devices:
MCP devices:
0005771479 [Microchip Technology Inc. MCP2200 USB Serial Port Emulator null] [id=0]
0007251135 [Microchip Technology Inc. MCP2200 USB Serial Port Emulator null] [id=1]
```

```
> java -jar DenkoviRelayCommandLineTool.jar ID=0 4v2 turn 1111
1111
> java -jar DenkoviRelayCommandLineTool.jar ID=0 4v2 all status
0000
```

```
> java -version
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-8u312-b07-1~deb9u1-b07)
OpenJDK Client VM (build 25.312-b07, mixed mode)
```

```
lsmod | grep cdc
dmesg | grep ttyACM
ls /dev/ttyACM*
/dev/ttyACM0
```
Shows the boards are attached and recognised.
Maybe #570, @PhoenixNP can tell a bit more how he got this working?

**Comment by theyosh** [comment, marked as answer]
Hmm, the only thin I know is that you need `sudo` to get it work. At least that is what I am doing in the code.

so does this make a change: `sudo java -jar 3rdparty/DenkoviRelayCommandLineTool/DenkoviRelayCommandLineTool.jar ID=0 4v2 1 1`

And maybe, first test with just 1 relay board connected?

But, does my TP software sees the relays? But they will not toggle?

And I made it work for PhoenixNP by using ssh access on his device. :P
